test/e2e/framework.go

```diff
@@ -887,7 +894,7 @@ func (data *TestData) restartCoreDNSPods(timeout time.Duration) error {
 	if err := data.clientset.CoreV1().Pods(antreaNamespace).DeleteCollection(context.TODO(), deleteOptions, listOptions); err != nil {
 		return fmt.Errorf("error when deleting all CoreDNS Pods: %v", err)
 	}
-	return data.waitForCoreDNSPods(timeout)
+	return retryOnConnectionLostError(retry.DefaultRetry, func() error { return data.waitForCoreDNSPods(timeout) })
```

Comment:
Is this new? Please add a comment to explain why it would happen or move it to another PR if this is not related

Reply:
Sure. Added comments at the function `retryOnConnectionLostError`.
This issue happens intermittently after antrea-agent restarted and AntreaIPAM enabled, because the Node uplink network is changed during antrea-agent starting/stopping.